lib/src/io_client.dart

```diff
@@ -72,3 +73,16 @@ class IOClient extends BaseClient {
     _inner = null;
   }
 }
+
+Uri _responseUrl(Request request, HttpClientResponse response) {
+  var redirects = response.redirects;
+
+  if (redirects.isEmpty) {
+    return request.url;
+  }
+
+  var location = redirects.last.location;
```

Comment:
Hypothetically if you called to `a.com/foo` that could redirect to another server `b.com/bar` which then does a relative redirect to `b.com/baz`.

I think I just need to run through the array with the `lastLocation` and use that to get the final url.